obspy/io/mseed/src/libmseed/unpackdata.c

```diff
@@ -119,7 +119,7 @@ msr_decode_float32 (float *input, int samplecount, float *output,
 
   for (idx = 0; idx < samplecount && outputlength > 0; idx++)
   {
-    sample = input[idx];
+    memcpy(&sample, &input[idx], sizeof(float));
 
```

Comment:
wow good call!

For me, it took a time to understand #485 again. And the nan reinterpretation is quite hidden here, as input in this function is `input` is `float *`, but gets a cast in pack.c from `int *`.

Good call!